### PR TITLE
[dev-launcher][plugin] Fix config plugin is not idempotent

### PR DESCRIPTION
**Pull request by lukmccall**
# Why

Closes ENG-1800.

# How

Updated the plugin checks to handle cases where someone doesn't use `;` or imported `dev-launcher` already.

# Test Plan

- create a new project & run prebuild.

### CODE REVIEW

File: packages/expo-dev-launcher/plugin/src/withDevLauncher.ts [hunk 1]
```diff
@@ -233,7 +234,10 @@ const withDevLauncherPodfile: ConfigPlugin = config => {
 const withErrorHandling: ConfigPlugin = config => {
   const injectErrorHandlers = async (config: ExportedConfigWithProps) => {
     await editIndex(config, index => {
-      if (!index.includes(DEV_LAUNCHER_JS_REGISTER_ERROR_HANDLERS)) {
+      if (
+        !index.includes(DEV_LAUNCHER_JS_REGISTER_ERROR_HANDLERS) &&
+        !index.includes(DEV_LAUNCHER_JS_REGISTER_ERROR_HANDLERS_VIA_LAUNCHER)
+      ) {
         index = DEV_LAUNCHER_JS_REGISTER_ERROR_HANDLERS + '\n\n' + index;
```
Comment (esamelson) [inline packages/expo-dev-launcher/plugin/src/withDevLauncher.ts:241]:
```suggestion
        index = DEV_LAUNCHER_JS_REGISTER_ERROR_HANDLERS + ';\n\n' + index;
```
To be consistent with the style of index.js in newly prebuilt projects, which does include semicolons.

**Review by esamelson — APPROVED** [review]

File: packages/expo-dev-launcher/plugin/src/withDevLauncher.ts (hunk 1 above)
Comment (lukmccall) [reply packages/expo-dev-launcher/plugin/src/withDevLauncher.ts:241]: Are you sure that we don't use semicolons in `index.js` anymore?

Comment (esamelson) [reply packages/expo-dev-launcher/plugin/src/withDevLauncher.ts:241]: I see them here https://github.com/expo/expo/blob/master/templates/expo-template-bare-minimum/index.js which is also what I see when I run expo init (blank) -> expo prebuild

Comment (lukmccall) [reply packages/expo-dev-launcher/plugin/src/withDevLauncher.ts]: Sorry, I for some reason thought that I should remove the semicolon not add it 🤦‍♂️ You are right. Nice find 🥇

**Pull request merged**